core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -0,0 +1,323 @@
+  // property and just use TEST_MONGODB?
+  public static final CalciteSystemProperty<Boolean> INTEGRATION_TEST =
+      booleanProperty("calcite.integrationTest", false);
+
+  /**
+   * Which database to use for tests that require a JDBC data source.
+   *
+   * The property can take one of the following values:
+   * <ul>
+   *   <li>HSQLDB(default)</li>
+   *   <li>H2</li>
+   *   <li>MYSQL</li>
+   *   <li>ORACLE</li>
+   *   <li>POSTGRESQL</li>
+   * </ul>
+   * If the specified value is not included in the previous list the default is used.
+   *
+   * <p>We recommend that casual users use hsqldb, and frequent Calcite developers use MySQL.
+   * The test suite runs faster against the MySQL database (mainly because of the 0.1s versus 6s
+   * startup time). You have to populate MySQL manually with the foodmart data set, otherwise there
+   * will be test failures.</p>
+   * */
+  public static final CalciteSystemProperty<String> TEST_DB =
+      stringProperty("calcite.test.db", "HSQLDB",
+          ImmutableSet.of(
+              "HSQLDB",
+              "H2",
+              "MYSQL",
+              "ORACLE",
+              "POSTGRESQL"));
+
+  /**
+   * Path to the dataset file that should used for integration tests.
+   *
+   * If a path is not set, then one of the following values will be used:
+   * <ul>
+   *   <li>../calcite-test-dataset</li>
+   *   <li>../../calcite-test-dataset</li>
+   *   <li>.</li>
+   * </ul>
+   * The first valid path that exists in the filesystem will be chosen.
+   */
+  public static final CalciteSystemProperty<String> TEST_DATASET_PATH =
+      new CalciteSystemProperty<>("calcite.test.dataset", v -> {
+        if (v != null) {
+          return v;
+        }
+        final String[] dirs = {
+            "../calcite-test-dataset",
+            "../../calcite-test-dataset"
+        };
+        for (String s : dirs) {
+          if (new File(s).exists() && new File(s, "vm").exists()) {
+            return s;
+          }
+        }
+        return ".";
+      });
+  /**
+   * Whether to run slow tests.
+   */
+  public static final CalciteSystemProperty<Boolean> TEST_SLOW =
+      booleanProperty("calcite.test.slow", false);
+  /**
+   * Whether to run MongoDB tests.
+   */
+  public static final CalciteSystemProperty<Boolean> TEST_MONGODB =
+      booleanProperty("calcite.test.mongodb", true);
+  /**
+   * Whether to run Splunk tests.
+   *
+   * Disabled by default, because we do not expect Splunk to be installed and populated data set.
+   */
+  public static final CalciteSystemProperty<Boolean> TEST_SPLUNK =
+      booleanProperty("calcite.test.splunk", false);
+  /**
+   * Whether to run Druid tests.
+   */
+  public static final CalciteSystemProperty<Boolean> TEST_DRUID =
+      booleanProperty("calcite.test.druid", true);
+  /**
+   * Whether to run Cassandra tests.
+   */
+  public static final CalciteSystemProperty<Boolean> TEST_CASSANDRA =
+      booleanProperty("calcite.test.cassandra", true);
```

Comment:
@zabetak , it is sad that public-facing properties are intermingled with internal "test-specific" ones like `calcite.test.cassandra`

Reply:
I agree it is not great but it was the first step in order to gather everything. I didn't have much inspiration on where to put those only for tests.

Reply:
Let's be clear that the purpose of this class is to get all properties in one place.

That directly contradicts the goal to separate properties used in tests from properties that have other purposes. So, I declare that a non-goal.

Would it help if we clarified that this class is NOT a public API? It's a single place that Calcite developers can go to.